client/my-sites/domains/domain-management/name-servers/index.jsx

```diff
@@ -16,66 +16,61 @@ import CustomNameserversForm from './custom-nameservers-form';
 import WpcomNameserversToggle from './wpcom-nameservers-toggle';
 import IcannVerificationCard from 'my-sites/domains/domain-management/components/icann-verification/icann-verification-card';
 import DnsTemplates from './dns-templates';
-import notices from 'notices';
 import paths from 'my-sites/domains/paths';
 import VerticalNav from 'components/vertical-nav';
 import VerticalNavItem from 'components/vertical-nav/item';
 import * as upgradesActions from 'lib/upgrades/actions';
 import { WPCOM_DEFAULTS, isWpcomDefaults } from 'lib/domains/nameservers';
 import { getSelectedDomain } from 'lib/domains';
 import isEmpty from 'lodash/isEmpty';
-import { successNotice } from 'state/notices/actions'
+import { errorNotice, successNotice } from 'state/notices/actions';
 
-const NameServers = React.createClass( {
-	propTypes: {
+class NameServers extends React.Component {
+	static propTypes = {
 		domains: React.PropTypes.object.isRequired,
```

Comment:
I know this isn't something that this refactor introduced, but just wondering if we really need domains in this component. It looks like we're just using `domains.hasLoadedFromServer`. It might be unlikely, but wouldn't any other change in the entire domains object would make this re-render?

Maybe we should just pass in `domains.hasLoadedFromServer`?